service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/disk/DeleteAzureDiskStep.java

```diff
@@ -20,40 +19,30 @@
 public class DeleteAzureDiskStep implements Step {
   private static final Logger logger = LoggerFactory.getLogger(DeleteAzureDiskStep.class);
   private final AzureConfiguration azureConfig;
-  private final ResourceDao resourceDao;
   private final CrlService crlService;
-  private final AzureCloudContext azureCloudContext;
-
-  private final UUID workspaceId;
-  private final UUID resourceId;
+  private final ControlledAzureDiskResource resource;
 
   public DeleteAzureDiskStep(
-      AzureConfiguration azureConfig,
-      AzureCloudContext azureCloudContext,
-      CrlService crlService,
-      ResourceDao resourceDao,
-      UUID workspaceId,
-      UUID resourceId) {
+      AzureConfiguration azureConfig, CrlService crlService, ControlledAzureDiskResource resource) {
     this.crlService = crlService;
-    this.resourceDao = resourceDao;
-    this.azureCloudContext = azureCloudContext;
     this.azureConfig = azureConfig;
-    this.workspaceId = workspaceId;
-    this.resourceId = resourceId;
+    this.resource = resource;
   }
 
   @Override
   public StepResult doStep(FlightContext context) throws InterruptedException {
-    var wsmResource = resourceDao.getResource(workspaceId, resourceId);
-    var disk = wsmResource.castToControlledResource().castToAzureDiskResource();
+    final AzureCloudContext azureCloudContext =
+        context
+            .getWorkingMap()
+            .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
 
     ComputeManager computeManager = crlService.getComputeManager(azureCloudContext, azureConfig);
     var azureResourceId =
         String.format(
```

Comment:
Seems like an odd place for this knowledge to live. I'd put a method on the resource itself to return the fully qualified ID.

Reply:
Retrieving the cloud context is a database transaction. I don't want to hit the database on every step of a flight, so factoring the fetch into a step and passing the data through seemed reasonable.

We could directly attach the cloud context to the resources in ControlledResource class and fetch that (if applicable) whenever we fetch the resource. That might be a worthy (but good sized) refactor:
- we would not have to go looking for the project id for GCP resources
- we would not need the separate transaction to lookup the cloud context
- it would make clear the requirement to have the context in place for resource creation, because it would be in the resource object.

I'll make a ticket for this work.